clients/src/main/java/org/apache/kafka/clients/producer/internals/TransactionManager.java

```diff
@@ -230,22 +167,15 @@ public boolean isValid() {
         }
     }
 
-    public synchronized FutureTransactionalResult initializeTransactions() {
+    public synchronized TransactionalRequestResult initializeTransactions() {
         ensureTransactional();
         transitionTo(State.INITIALIZING);
-        setPidAndEpoch(NO_PRODUCER_ID, NO_PRODUCER_EPOCH);
+        setPidAndEpoch(PidAndEpoch.NONE);
         this.sequenceNumbers.clear();
-        if (transactionCoordinator == null)
-            pendingTransactionalRequests.add(findCoordinatorRequest(FindCoordinatorRequest.CoordinatorType.TRANSACTION, transactionalId, false));
```

Comment:
Why don't we need this any more? And why do we do `InitPidRequest` unconditionally?